rpcs3/Emu/Cell/Modules/cellMusic.cpp

```diff
@@ -2,131 +2,287 @@
 #include "Emu/System.h"
 #include "Emu/IdManager.h"
 #include "Emu/Cell/PPUModule.h"
-
-#include "cellMusic.h"
+#include "Emu/Cell/lv2/sys_lwmutex.h"
+#include "Emu/Cell/lv2/sys_lwcond.h"
+#include "Emu/Cell/lv2/sys_spu.h"
+#include "cellSearch.h"
+#include "cellSpurs.h"
 #include "cellSysutil.h"
 
-
+#include "cellMusic.h"
 
 logs::channel cellMusic("cellMusic");
 
-struct music2_t
+struct music_t
 {
-	vm::ptr<CellMusic2Callback> func;
+	vm::ptr<void(u32 event, vm::ptr<void> param, vm::ptr<void> userData)> func;
 	vm::ptr<void> userData;
 };
 
-s32 cellMusicGetSelectionContext()
+s32 cellMusicGetSelectionContext(vm::ptr<CellMusicSelectionContext> context)
 {
-	UNIMPLEMENTED_FUNC(cellMusic);
+	cellMusic.todo("cellMusicGetSelectionContext(context=*0x%x)", context);
 	return CELL_OK;
 }
 
-s32 cellMusicSetSelectionContext2()
+s32 cellMusicSetSelectionContext2(vm::ptr<CellMusicSelectionContext> context)
 {
-	UNIMPLEMENTED_FUNC(cellMusic);
+	cellMusic.todo("cellMusicSetSelectionContext2(context=*0x%x)", context);
+
+	const auto music = fxm::get_always<music_t>();
+
+	if (!music->func)
+		return CELL_MUSIC2_ERROR_GENERIC;
+
+	sysutil_register_cb([=](ppu_thread& ppu) -> s32
+	{
+		music->func(ppu, CELL_MUSIC2_EVENT_SET_SELECTION_CONTEXT_RESULT, vm::addr_t(CELL_OK), music->userData);
```

Comment:
I am not sure if `vm::addr_t(CELL_OK)` is the right thing to do here. `vm::null` could also be a solution.
This value can be casted to an `uint32_t` inside the callback before usage.

Reply:
Jojo is still working fine, although I've got some freezes from time to time (memory access).
TekkenTagTournament2 goes ingame and can be played, but it has more glitches (no bgm for example). Freezes randomly too.
The 2 log files:
[RPCS3.log.JoJo2.gz](https://github.com/RPCS3/rpcs3/files/1744993/RPCS3.log.JoJo2.gz)
[RPCS3.log-Tekken2.gz](https://github.com/RPCS3/rpcs3/files/1744995/RPCS3.log-Tekken2.gz)
Nice to see some progress for these games.